### DISCUSSION TITLE
kyuubi连接池 hive数据源 sql task 连接太多

### DISCUSSION THREAD

**Discussion by yqf1991** [apache/dolphinscheduler#12286] [Q&A] [answered]
worker-server/conf/common.properties
spring.datasource.maxActive=3
support.hive.oneSession=true:

然后把HikariPool的日志打开,惊人发现,配置的hive数据源(后台是kyuubi),会启动多个连接池,每个连接池都是3个,这样,总共就能有100多个连接了,很快就会把kyuubi的内存打爆,
为什么一个数据源,sql task的执行,会启动维护多个连接池,如果没上限的启动连接池,再多内存的hive server都不够用啊
![image](https://user-images.githubusercontent.com/14036137/194791662-1335d46b-ede5-4785-9aa0-da149adc9c22.png)

**Comment by DarkAssassinator** [comment, marked as answer]
may u can check this question: https://github.com/apache/dolphinscheduler/discussions/12213
